src/AnalysisDriver.cpp

```diff
@@ -0,0 +1,11 @@
+#include "AnalysisDriver.h"
+
+AnalysisDriver::AnalysisDriver(const Configuration &configuration)
+    : configuration_{configuration},
+      eval_expression_analysis_{
+          std::make_unique<EvalExpressionAnalysis>(configuration)} {}
```

Comment:
why not just `eval)expression_analysis_{configuration}`

Reply:
`unique_ptr` was a later addition when I was debugging and suspected that the class was being deleted somehow. That assumption turned out to be false but I let the `unique_ptr` linger on.
I think we are not gaining much from `unique_ptr` as of now. I will remove it.

Reply:
I don't understand the advantage of unique pointer here. Since we will have a copy of configuration in each class, than we do not need it, right?